app/src/main/java/com/kabouzeid/gramophone/service/MediaButtonIntentReceiver.java

```diff
@@ -133,13 +133,18 @@ public static boolean handleIntent(final Context context, final Intent intent) {
 
                         // The service may or may not be running, but we need to send it
                         // a command.
-                        if (keycode == KeyEvent.KEYCODE_HEADSETHOOK) {
+                        int keycodeToCompare = KeyEvent.KEYCODE_HEADSETHOOK;
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                            keycodeToCompare = KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE;
```

Comment:
This isn't actually new to O, so you should just check if it's `KEYCODE_HEADSETHOOK` or `KEYCODE_MEDIA_PLAY_PAUSE`.

Reply:
it isn't new but it seems that using headphones through usb (with the connector jack-usb) the event is detected as `KEYCODE_MEDIA_PLAY_PAUSE` and if it is plugged through audio jack connector it is detected as `KEYCODE_HEADSETHOOK`, so the solution is to see if the phone has or not jack connector and compare to the correct keycode (i don't know what keycode bluetooth headphones send, but i suppose it is `KEYCODE_HEADSETHOOK`) or just change the keycode to `KEYCODE_HEADSETHOOK` if a `KEYCODE_MEDIA_PLAY_PAUSE` is present before the comparations.

Reply:
I'm not sure if it really matters though in either case. You're reacting to the keycodes in the same way, and it's not like both will happen and you don't want one to react, right?

Reply:
Yes, comparing if the keycode is one of both works fine